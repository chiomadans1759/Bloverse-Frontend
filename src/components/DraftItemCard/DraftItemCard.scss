@import 'sass-rem';

.bl-draft-item {
  border-bottom: 1px solid #dee2e6 !important;
  border-top: 1px solid #dee2e6 !important;
  border-left: none !important;
  border-right: none !important;
  border-radius: 0 !important;
  background: #ffffff;
  margin-bottom:rem(4px);
  &:first-child {
    margin-top: 0 !important;
  }
  &-body {
    background: #ffffff;
    .title {
      margin: rem(12px) auto;
      margin-top: 0;
      height: auto;
      &--text {
        padding: 0;
        height: 100%;
        margin: auto;
        flex-direction: column;
        &>button {
          border-color: #dee2e6;
          &:hover {
            color: #2F80ED;
          }
        }
        &>span {
          max-width: none;
          width: 100%;
          font-size: rem(14px);
          text-align: left;
          color: #868686;
          text-transform: capitalize;
          padding: 0;
        }
        a{
          color: #333333;
        }
        .description {
          max-width: none;
          width: 100%;
          height: rem(40px);
          font-size: rem(14px);
          text-align: left;
          color: #252525;
          padding: 0;
          padding-right: rem(16px);
          padding-bottom: rem(5px);
          overflow: hidden; 
        }
        .username {
          display: flex;
          align-items: center;
          max-width: none;
          width: 100%;
          text-transform: capitalize;
          padding: 0;
          font-size: rem(14px);
          color: #565656;
          .initials {
            height: rem(16px);
            width: rem(16px);
            font-size: rem(8px);
            text-align: center;
            color: #ffffff;
            padding: 0;
            padding-top: rem(2.5px);
            padding-bottom: rem(1.5px);
            background-color: #8C40FF;
            border: none;
          }
          .name {
            color: #565656;
            font-weight: 500;
            font-size: rem(14px);
          }
          .name::after {
            content: '\2022';
            color: #2F80ED;
            margin-left: rem(5px);
            font-weight: 800;
          }
          .time {
            font-size: rem(12px);
            color: #868686;
          }
        }
      }
      &--image {
        display: inline-flex;
        justify-content: flex-start;
        background-repeat: no-repeat;
        background-size: cover;
        background-position: center;
        height: rem(52px);
        width: rem(62px);
      }
    }
    .keypoints {
      margin: rem(12px) auto;
      max-height: rem(150px);
      overflow: hidden;
      span {
        font-size: rem(14px);
        color: #252525;
        margin: rem(10px) auto;
      }
      &--lists {
        padding: 0 rem(16px);
        white-space: nowrap;
        word-wrap: break-word;
        overflow: hidden;
        text-overflow: ellipsis;
        color: #565656;
        list-style: none;
        li:first-child {
          margin-top: rem(5px);
        }
        li::before {
          content: "\2022";
          color: #2F80ED;
          font-size: rem(14px);
          font-weight: 800;
          display: inline-block; 
          width: 1em;
          margin-left: -1em;
        }
      }
    }
    .actions {
      height: rem(21px);
      margin: rem(12px) auto;
      margin-bottom: 0;
      width: 100%;
      .count {
        margin: rem(7px);
        color: #252525;
      }
      .views {
        color: #868686;
        padding: 0;
        cursor: pointer;
        font-size: rem(14px);
      }
      .others {
        margin:0;
        padding: 0;
        .likes {
          border-right: rem(1px) solid #e6e5e5;
          padding: 0;
          display: inline-flex;
          justify-content: space-around;
          color: #2F80ED;
          font-size: rem(14px);
          .thumbs-up, .thumbs-down {
            cursor: pointer;
          }
        }
        .comments {
          border-right: rem(1px) solid #e6e5e5;
          padding: 0;
          display: inline-flex;
          justify-content: space-around;
          align-items: center;
          color: #868686;
          .coment, .share {
            display: none;
            cursor: pointer;
          }
          .ellipsis {
            cursor: pointer;
            font-size: rem(25px);
            .dropdown-item{
              font-size: rem(17px);
              padding:.55rem 1.5rem;
              svg{
                color: #096DD9 !important;
              }
              .badges{
                float: right;
                color: #096DD9;
                border: rem(1px) solid #096DD9;
                border-radius: rem(3px);
                padding: 0 rem(3px);
              }
            }
            .dropdown-toggle {
              #option-icon {
                display: inline-block;
                background-color: #bcbcbc;
                width: rem(3px);
                height: rem(3px);
                margin-left: rem(2px);
                border-radius: 50%;
              }
              &::after {
                display: none;
              }
            }
          }
        }
        .toggle-keypoints {
          padding: 0;
          display: inline-flex;
          justify-content: flex-end;
          align-items: center;
          font-size: rem(25px);
          color: #868686;
          cursor: pointer;
        }
      }
    }

    .actions .others .comments .ellipsis {
      .dropdown-menu {
        &__categories {
          width: rem(335px);
          max-height: rem(341px);
          overflow-y: scroll;
          overscroll-behavior-y: smooth;
          flex-direction: column;
          margin: 0;
          .favourite-categories {
            width: 100%;
            padding: rem(12px) 0;
            border-bottom: rem(1px) solid rgba(0, 0, 0, 0.10);
            .header {
              font-size: rem(14px);
              color: #B7B7B7;
              text-transform: capitalize;
              padding: rem(4px) rem(16px);
            }
            &-list {
              margin: 0;
              &>span {
                white-space: nowrap; 
                overflow: hidden;
                text-overflow: ellipsis;
              }
            }
          }
          .other-categories {
            width: 100%;
            padding: rem(12px) 0;
            .header {
              font-size: rem(14px);
              text-transform: capitalize;
              color: #B7B7B7;
              padding: rem(4px) rem(16px);
              white-space: nowrap; 
              overflow: hidden;
              text-overflow: ellipsis;
            }
            &-list {
              margin: 0;
              &>span {
                white-space: nowrap; 
                overflow: hidden;
                text-overflow: ellipsis;
              }
            }
          }
        }
      }
    }
  }
  &-footer {
    .actions {
      .edit, .delete {
        cursor: pointer;
        .icons {
          color: #2F80ED;
          height: rem(28px);
          width: rem(28px);
          display: inline-flex;
          justify-content: center;
          align-items: center;
          .fa-thumbs-down {
            transform: rotateY(180deg);
          }
        }
      }
    }
  }
}

.hideCardBorder {
  border: none !important;
}

@media screen and (min-width: 760px) {
  .bl-draft-item {
    &-body {
      .title {
        height: rem(100px);
        &--text {
          .description {
            font-size: rem(11px);
            overflow: hidden;
          }
        }
      }
      .actions .others .comments {
        .coment, .share {
          display: inline-block;
        }
      }
    }
  }
}

@media screen and (min-width: 1020px) {
  .bl-draft-item {
    border: rem(1px) solid #d9d9d9;
    border-radius: 0 !important;
    background: #ffffff;
    margin-bottom:rem(8px);
    &-body {
      background: #ffffff;
      .title {
        margin: rem(12px) auto;
        margin-top: 0;
        height: rem(125px);
        &--text {
          padding: 0;
          height: 100%;
          margin: auto;
          flex-direction: column;
          &>span {
            max-width: none;
            width: 100%;
            font-size: rem(16px);
            text-align: left;
            color: #868686;
            text-transform: capitalize;
            padding: 0;
          }
          .description {
            max-width: none;
            width: 100%;
            height: rem(56px);
            font-size: rem(20px);
            text-align: left;
            color: #252525;
            padding: 0;
            padding-right: rem(16px);
            padding-bottom: rem(5px); 
          }
          .username {
            display: flex;
            align-items: center;
            max-width: none;
            width: 100%;
            text-transform: capitalize;
            padding: 0;
            font-size: rem(16px);
            color: #565656;
            .initials {
              height: rem(24px);
              width: rem(24px);
              font-size: rem(10px);
              text-align: center;
              padding: 0;
              padding-top: rem(2.5px);
              padding-bottom: rem(1.5px);
              background-color: #8C40FF;
              border: none;
            }
            .name {
              color: #565656;
              font-weight: 500;
              font-size: rem(16px);
            }
            .name::after {
              content: '\2022';
              color: #2F80ED;
              margin-left: rem(5px);
              font-weight: 800;
            }
            .time {
              font-size: rem(14px);
              color: #868686;
            }
          }
        }
        &--image {
          display: inline-flex;
          justify-content: flex-start;
          background-repeat: no-repeat;
          background-size: cover;
          background-position: center;
          height: rem(118px);
          .feed-image {
            justify-content: flex-end;
            width: 100%;
            height: 100%;
          }
        }
      }
      .keypoints {
        margin: rem(12px) auto;
        max-height: rem(150px);
        overflow: hidden;
        transition: all, 150s ease-in-out;
        span {
          font-size: rem(16px);
          color: #252525;
          margin: rem(10px) auto;
        }
        &--lists {
          padding: 0 rem(16px);
          white-space: nowrap;
          word-wrap: break-word;
          overflow: hidden;
          text-overflow: ellipsis;
          color: #565656;
          list-style: none;
          li:first-child {
            margin-top: rem(5px);
          }
          li::before {
            content: "\2022";
            color: #2F80ED;
            font-size: rem(16px);
            font-weight: 800;
            display: inline-block; 
            width: 1em;
            margin-left: -1em;
          }
        }
      }
      .actions {
        height: rem(21px);
        margin: rem(12px) auto;
        margin-bottom: 0;
        .count {
          margin: rem(7px);
          color: #252525;
        }
        .views {
          color: #868686;
          padding: 0;
          cursor: pointer;
        }
        .others {
          margin:0;
          padding: 0;
          .likes {
            border-right: rem(1px) solid #e6e5e5;
            padding: 0;
            display: inline-flex;
            justify-content: space-around;
            color: #2F80ED;
            .thumbs-up, .thumbs-down {
              cursor: pointer;
            }
          }
          .comments {
            border-right: rem(1px) solid #e6e5e5;
            padding: 0;
            display: inline-flex;
            justify-content: space-around;
            align-items: center;
            color: #868686;
            .coment, .share {
              display: inline-block;
              cursor: pointer;
            }
            .ellipsis {
              cursor: pointer;
              font-size: rem(20px);
            }
          }
          .toggle-keypoints {
            padding: 0;
            display: inline-flex;
            justify-content: flex-end;
            align-items: center;
            font-size: rem(25px);
            color: #868686;
            cursor: pointer;
          }
        }
      }
    }
  }
}

@media screen and (max-width: 576px) {
  .bl-draft-item {
    &-body {
      .title {
        height: auto !important;
      }
    }
  }
}

::-webkit-scrollbar {
  display: none;
};

.svg-inline--fa.fa-w-14, .fa-edit {
  color: #2F80ED;
}
