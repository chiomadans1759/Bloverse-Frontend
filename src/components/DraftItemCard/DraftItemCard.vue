<template>
  <main id="bl-draft-item" class="card bl-draft-item">
    <div class="card-body bl-draft-item-body">
      <div class="card-title row title">
        <div class="col-xs-12 col-sm-12 col-md-9 row title--text">
          <span class="col-3">category</span>
          <p class="col-5 col-sm-6 font-weight-bold description">
            <router-link to="/posts/rere">
              Lorem ipsum dolor sit amet, vel accumsan iberaviss extttttt, ea nec elaboraret interpret
              Lorem ipsum dolor sit amet, vel accumsan iberaviss ex, ea nec elaboraret interpret
              Lorem ipsum dolor sit amet, vel accumsan iberaviss ex, ea nec elaboraret interpret
            </router-link>
          </p>
          <p class="col-4 col-sm-3 username d-none d-sm-block">
            <span class="time">16 hours ago</span>
          </p>
        </div>
        <div class="d-none col-3 d-md-flex justify-content-around align-items-center flex-wrap">
          <div>
            <a href="#">
              <div class="text-center"><i class="fas fa-edit"></i></div>
              <div>Edit</div>
            </a>
          </div>
          <div>
            <a href="#">
              <div class="text-center"><i class="fas fa-trash-alt"></i></div>
               <div>Delete</div>
            </a>
          </div>
        </div>
      </div>
      <div class="d-md-none card-text row actions justify-content-between">
        <div class="col-5 col-sm-0 views d-block d-sm-none">
          <span class="time">16 hrs ago</span>
        </div>
        <div class="col-7 col-sm-12 row others justify-content-end">
          <p class="mr-3">
            <i class="fal fa-edit"></i>
            <span class="text d-none d-sm-inline">Edit</span>
          </p>
          <p class="delete">
            <i class="fal fa-trash-alt"></i>
            <span class="text d-none d-sm-inline">Delete</span>
          </p>
        </div>
      </div>
    </div>
  </main>
</template>

<script>

export default {
  name: "PostCard",
};
</script>

<style lang="scss" scoped>
@import "./DraftItemCard.scss";
</style>
