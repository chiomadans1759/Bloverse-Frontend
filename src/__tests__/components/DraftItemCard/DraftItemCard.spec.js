import { mount, RouterLinkStub  } from '@vue/test-utils'
import DraftItemCard from '@/components/DraftItemCard/DraftItemCard.vue';

describe('DraftItemCard Component', () => {
  it('should render correctly', () => {

    const wrapper = mount(DraftItemCard , {
      stubs: {
        RouterLink: RouterLinkStub
      }
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
