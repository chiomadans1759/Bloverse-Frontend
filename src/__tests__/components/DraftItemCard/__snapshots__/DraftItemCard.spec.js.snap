// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DraftItemCard Component should render correctly 1`] = `
<main
  class="card mt-4 bl-draft-item"
  id="bl-draft-item"
>
  <div
    class="card-body bl-draft-item-body"
  >
    <div
      class="card-title title"
    >
      <div
        class="title--text"
      >
        <button
          class="btn btn-outline-primary btn-sm bg-white mb-3"
          type="button"
        >
          Category
        </button>
         
        <p
          class="font-weight-bold description"
        >
          <a>
            
            Lorem ipsum dolor sit amet, vel accumsan iberaviss extttttt, ea nec elaboraret interpret
            Lorem ipsum dolor sit amet, vel accumsan iberaviss ex, ea nec elaboraret interpret
            Lorem ipsum dolor sit amet, vel accumsan iberaviss ex, ea nec elaboraret interpret
          
          </a>
        </p>
         
        <p
          class="col-4 col-sm-3 username d-none d-sm-block mt-3"
        >
          <span
            class="time"
          >
            1 min ago
          </span>
        </p>
      </div>
    </div>
  </div>
   
  <div
    class="border-top bl-draft-item-footer"
  >
    <div
      class="d-flex justify-content-end px-4 py-2 actions"
    >
      <p
        class="mx-3 edit"
      >
        <span
          class="border rounded-circle icons"
        >
          <i
            class="fal fa-edit"
          />
        </span>
      </p>
       
      <p
        class="delete"
      >
        <span
          class="border rounded-circle icons"
        >
          <i
            class="fal fa-times deleteIcon"
          />
        </span>
      </p>
    </div>
  </div>
</main>
`;
